python/ray/util/sgd/v2/trainer.py

```diff
@@ -374,35 +425,48 @@ class SGDIterator:
     def __init__(
             self, backend_executor: BackendExecutor,
             train_func: Union[Callable[[], T], Callable[[Dict[str, Any]], T]],
-            checkpoint: Optional[Dict],
-            checkpoint_strategy: Optional[CheckpointStrategy]):
+            checkpoint: Optional[Union[Dict, str, Path]],
+            checkpoint_strategy: Optional[CheckpointStrategy], run_dir: Path):
         self._executor = backend_executor
         self._train_func = train_func
         self._checkpoint_strategy = checkpoint_strategy
-        self._start_training(train_func, checkpoint, checkpoint_strategy)
+        self._run_dir = run_dir
+        self._start_training(
+            train_func, checkpoint, checkpoint_strategy, run_dir=run_dir)
 
         self._final_results = None
         self._finished_training = False
 
     def __iter__(self):
         return self
 
-    def _start_training(self, train_func, checkpoint, checkpoint_strategy):
+    def _start_training(self,
+                        train_func,
+                        checkpoint,
+                        checkpoint_strategy,
+                        run_dir,
+                        checkpoint_id=None):
```

Comment:
For consistency let's just name everything `latest_checkpoint_id` for now.

```suggestion
                        latest_checkpoint_id=None):
```